### PR TITLE
Don't use disabled in job defs

### PR DESCRIPTION
**Pull request by benvand**
Remove explicitly setting disabled in the jobs, this overrides the disabled flag set in job_definitions/_defaults.yml which is bad

This flag should not be overrider as jobs are disabled when jenkins is built to prevent unexpected runs

### CODE REVIEW

**Comment by lfdebrux** [comment]
Hmm, that is a good point... However it is annoying 😆 How useful is it to be able to toggle disabled like this? Thoughts @katstevens?

**Comment by katstevens** [comment]
Hmm. It's pretty useful for when we're trying to setup new framework jobs in advance. For example the performance platform jobs will fail if the framework isn't open for applications (this is probably a good thing - we should know about it if the framework status isn't `open` for whatever reason). We could potentially change the scripts themselves to be more robust but that'll need some work.

**Comment by lfdebrux** [comment]
Okay I've added some commits that I think make a reasonable workaround

File: job_definitions/stats_snapshots.yml [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-    name: 'snapshot-stats-{{ framework }}-{{ environment }}'
-    disabled: {{ disabled }}
-    display-name: "Snapshot {{ framework }} stats - {{ environment }}"
-    project-type: freestyle
+    name: stats-snapshot-{{ framework }}-{{ environment }}
+    display-name: Snapshot stats for {{ framework }} - {{ environment }}
     description: Creates an hourly snapshot of framework stats and stores it in the API audit event
+    {% if not enabled %}
+    disabled: true
+    {% endif %}
```
Comment (katstevens) [inline job_definitions/stats_snapshots.yml]: I think this should be fine. Though be careful with whitespace/indentation of `{% for %}` loops in Jenkins jobs templates, we've come unstuck with those before 😿

Comment (lfdebrux) [reply job_definitions/stats_snapshots.yml]: Eep good point

Comment (lfdebrux) [reply job_definitions/stats_snapshots.yml]: I'll change it

**Comment by benvand** [comment]
@katstevens I can't approve this either :/

File: job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml [hunk 2]
```diff
@@ -21,60 +16,60 @@
...
+            - staging
+            - production
+      - string:
+          name: FRAMEWORK
     dsl: |
       node {
           try {
-            stage('Prepare') {
-              dir('scripts/generated-countersignature-pdfs') {
-                deleteDir()
```
Comment (katstevens) [inline job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml]: Are we no longer explicitly cleaning up this folder? We have a new data folder cleaned & created below - is that just for cloning repos? Where do the PDFs go?

**Comment by katstevens** [comment]
I am happy with the changes to the stats jobs, but the generate-and-upload job is more of a major change (and is an essential job for G11 suppliers right now). Is there a way we can test it?

File: job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml (hunk 2 above)
Comment (lfdebrux) [reply job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml]: PDFs go to `data/generated-countersignature-pdfs`, so it gets cleaned up by the `rm -rf` below

**Comment by lfdebrux** [comment]
> I am happy with the changes to the stats jobs, but the generate-and-upload job is more of a major change (and is an essential job for G11 suppliers right now). Is there a way we can test it?

I've added a dry-run flag and am going to test now by uploading a preview version of the job to Jenkins.

**Comment by lfdebrux** [comment]
Have tested the counterpart signatures job on Jenkins, works now

File: job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml [hunk 3]
```diff
@@ -96,6 +97,34 @@
...
+    display-name: 'Generate, upload and notify counterpart signature pages for {{', '.join(s) for s,_,enabled if enabled}}'
+    project-type: pipeline
+    description: |
+        This job will run a series of scripts to
+        <ol>
+          <li>Generate framework agreement counterpart signature pages for suppliers approved for countersignature but
+              without a countersigned agreement</li>
+          <li>Upload the generated files to S3 and set the path in the relevant FrameworkAgreement record</li>
+          <li>Send an email to all active users on the supplier account that the counterpart file has been uploaded</li>
+        </ol>
```
Comment (katstevens) [inline job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml]: We have the same description on this `-periodic` job and the one above - we should probably mention that this one calls the other?

File: job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml [hunk 4]
```diff
@@ -96,6 +97,34 @@
             ]
           }
       }
+  
+- job:
+    name: 'generate-upload-and-notify-counterpart-signature-pages-periodic'
+    display-name: 'Generate, upload and notify counterpart signature pages for {{', '.join(s) for s,_,enabled if enabled}}'
```
Comment (katstevens) [inline job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml]: Can we include the `<stage>` in the `display-name` if it's not a parameter we can set when building the job?

File: job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml (hunk 2 above)
Comment (katstevens) [reply job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml]: Aren't we deleting `scripts/generated-countersignature-pdfs` in the `Cleanup` stage?

Comment (lfdebrux) [reply job_definitions/generate_upload_and_notify_counterpart_signature_pages.yml]: Ah good spot!!

**Comment by lfdebrux** [comment]
Okay I think I've ironed out all the bugs?

**Comment by katstevens** [comment]
Just thought of another thing (sorry! 😹) Will these new jobs need to be included in the config list to show up on the Framework Lifecycle jobs tab?

**Comment by lfdebrux** [comment]
> Just thought of another thing (sorry! ) Will these new jobs need to be included in the config list to show up on the Framework Lifecycle jobs tab?

Done

**Comment by lfdebrux** [comment]
> I am finally happy 😸

Thanks for being so diligent!

**Pull request merged**